### ISSUE TITLE
False positive UnnecessaryLet rule

### ISSUE THREAD

**Issue by arthur-ghazaryan** [detekt/detekt#4229]
## Expected Behavior
The code below should not trigger `UnnecessaryLet` rule violation, because `return` is used to return from `test` function
```kotlin
    private var nullableIntent : Intent? = null
    private var nonNullIntent : Intent = Intent("action")

    private fun test() : Intent {
        nullableIntent?.let { return it }

        return nonNullIntent
    }
```

## Observed Behavior
Triggers `UnnecessaryLet` 

## Your Environment

* Version of detekt used: 1.19.0-RC1
* Version of Gradle used (if applicable):  7.0.1
